### PR TITLE
[#112422463] Know immediately if I am successful in my framework application on an individual framework's (dashboard) page

### PR DESCRIPTION
**Pull request by pcraig3**
This pull request completes the [DOS framework agreements page](https://www.pivotaltracker.com/story/show/112422463) story. The first part of that story had to do with uploading/downloading the framework agreements, so this the "know immediately if I am successful" part of it.

Changed a few of the messages on the frameworks dashboard such that:
- If you've been allowed onto the framework, it should be abundantly clear
- If not, we stick with the current messaging which doesn't really say anything either way.  Basically it's: 'Here's your letter. Read it.'

And then -- because I found the whole thing super confusing -- I've broken the template up into smaller ones like we've done in the [supplier template](https://github.com/alphagov/digitalmarketplace-supplier-frontend/blob/master/app/templates/suppliers/dashboard.html#L42).

[>> Story on Pivotal](https://www.pivotaltracker.com/story/show/112422463)

***

### Accepted to framework (in standstill)
screen shot 2016-02-09 at 14 42 23

### Not accepted to framework (in standstill)
screen shot 2016-02-09 at 14 42 40

*** 

Here's all of the various states for the frameworks dashboard page for a supplier who is/has applied to the framework.  Page is dependent on (a) framework status and (b) supplier actions.

[Open](https://cloud.githubusercontent.com/assets/2454380/12919014/17cc635e-cf3a-11e5-8d8c-6cb0e7fbb490.png)
[Pending](https://cloud.githubusercontent.com/assets/2454380/12919016/17d78e46-cf3a-11e5-8f78-ca18d34f6b04.png)
[Standstill + not on framework](https://cloud.githubusercontent.com/assets/2454380/12918931/c052f5f2-cf39-11e5-8f41-7a9ee64361c6.png)
[Standstill + on framework](https://cloud.githubusercontent.com/assets/2454380/12918934/c1def024-cf39-11e5-9bc6-258ed04dcfcd.png)
[Standstill + returned countersigned agreement](https://cloud.githubusercontent.com/assets/2454380/12919015/17d0199a-cf3a-11e5-9bf7-30d726f56be4.png)
[Live + not on framework](https://cloud.githubusercontent.com/assets/2454380/12919076/50316352-cf3a-11e5-9675-62146947595f.png)
[Live + on framework](https://cloud.githubusercontent.com/assets/2454380/12919075/502847f4-cf3a-11e5-855e-2ce1063317de.png)
[Live + returned countersigned agreement](https://cloud.githubusercontent.com/assets/2454380/12945095/299652ee-cfe3-11e5-92dd-2e544e12c45a.png)

### CODE REVIEW

File: app/templates/frameworks/_dashboard_lede.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {% elif framework.status == 'pending' %}
+      {# supplier_is_on_framework should never be true before standstill #}
+      <p>You made your supplier declaration and submitted {{ counts.complete }} {{ 'service' if counts.complete == 1 else 'services' }} for consideration.</p>
+      <p>A letter informing you whether your application was successful or not will be posted on your {{ framework.name }} updates page by {{ dates.intention_to_award_date|safe }}.</p>
+
+    {% elif framework.status == 'standstill' %}
+      {% if supplier_is_on_framework %}
+        <p>Your application was successful. You'll be able to sell services when the {{ framework.name }} framework is live.</p>
+      {% else %}
+        <p>You made your supplier declaration and submitted {{ counts.complete }} {{ 'service' if counts.complete == 1 else 'services' }}.</p>
```
Comment (robyoung) [inline app/templates/frameworks/_dashboard_lede.html]: We could be more explicitly negative if `supplier_is_on_framework` is `False` rather than `None`.

**Comment by robyoung** [comment]
Has this been tested with :sparkles: functional tests :sparkles:? And have any new scenarios been added?

File: app/templates/frameworks/_framework_actions.html [hunk 2]
```diff
@@ -0,0 +1,110 @@
+{% if framework.status in ['standstill', 'live'] and application_made and not countersigned_agreement_file %}
+<li class="browse-list-item">
+  <a class="browse-list-item-link" href="{{ url_for('.download_agreement_file', framework_slug=framework.slug, document_name='result-letter.pdf') }}" download>
+    <span>Download your application {% if supplier_is_on_framework %}award{% else %}result{% endif %} letter (.pdf)</span>
+  </a>
+  {% if supplier_is_on_framework %}
+    <p>This letter is a record of your successful {{ framework.name }} application.</p>
+  {% else %}
+    <p>This letter informs you if your {{ framework.name }} application has been successful.</p>
```
Comment (robyoung) [inline app/templates/frameworks/_framework_actions.html]: This seems weird. If they're successful we tell them, if they're not we wave our hands a bit.

Comment (pcraig3) [inline app/templates/frameworks/_framework_actions.html]: Yeah, @ralph-hawkins had a quote about not breaking up with people over text message.

Deliver bad news in a serious way, basically; don't just have a cheesy on-screen message, but let the official letter be the serious thing.

Also, Ralph, Roz and Cath have all looked at it and they're happy.

File: app/templates/frameworks/_dashboard_lede.html (hunk 1 above)
Comment (pcraig3) [inline app/templates/frameworks/_dashboard_lede.html]: Sure :smile:

**Comment by pcraig3** [comment]
@robyoung I've tested that the existing functional tests work.  
Didn't add any new scenarios because it seems like a lot of moving parts (ie, supplier has to have applied to a framework which is in a specific state for each on-screen message).  
Discussed it with Andrew about it earlier and he agreed.

Do you think we need them?

**Comment by robyoung** [comment]
Sounds good to me :+1:

**Pull request merged**